### PR TITLE
migrated from npm to yarn

### LINKED ISSUE #312: migration from npm to Yarn
# Proposal: Migrate from npm to Yarn

## Summary

Considering cms growth, switching from npm to Yarn could bring significant advantages in terms of performance, reliability, and developer experience. 

Additionally, the use of different package managers such as npm, Yarn, and pnpm by contributors when making pull requests can lead to conflicts, complicating the development workflow and LOCK files.
##
**I would like to work on this issue and fix this**

### PR DESCRIPTION
**Pull request by rohithreddy009**
### PR Fixes:
- 1 fixes #312 

Resolves #[312] 

### Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] I assure there is no similar/duplicate pull request regarding same issue
- [ ] migrated from npm to yarn using yarn official docs https://classic.yarnpkg.com/lang/en/docs/migrating-from-npm/
- [ ] changed readme.md file to setup the project using yarn
- [ ] changed the docker file to use yarn instead of npm
- [ ] ensured everything is working perfect in local development
Screenshot 2024-04-02 at 3 03 05 PM

https://github.com/code100x/cms/assets/89250524/81d932eb-d49b-4c21-b570-5fec4f9aec0e

Screenshot 2024-04-02 at 2 31 07 PM

### CODE REVIEW

**Comment by rohithreddy009** [comment]
@hkirat sir, done migrating from npm to yarn.
If anything is missing let me know

**Comment by rohithreddy009** [comment]
**I have only changed Dockerfile, README file for stating how to setup the project locally using yarn to contributors, deleted package-lock.json and pnpm-lock.yaml**

**The changes in the other components are autoformatted by prettier, I haven't done any manual changes**

**Comment by ssk090** [comment]
Hey @rohithreddy009, I just wanted to know how much time does the `yarn` command takes to install all the dependencies locally ?

**Comment by rohithreddy009** [comment]
> Hey @rohithreddy009, I just wanted to know how much time does the `yarn` command takes to install all the dependencies locally ?

Depends on your internet. For me 3-5 minutes. I'm on 100mbps internet.

**Comment by rohithreddy009** [comment]
I'm saying, if you forked the repo and downloading packages for the first time.

**Comment by ssk090** [comment]
> I'm saying, if you forked the repo and downloading packages for the first time.

https://github.com/rohithreddy009/cms/pull/1
check this PR

Results:
image

**Comment by Neon-20** [comment]
Do we really need to move from npm to yarn atm? 👀

**Comment by rohithreddy009** [comment]
> Do we really need to move from npm to yarn atm? 👀

I think this is the correct time to migrate, yarn has many advantages and faster than npm. If this is prolonged it becomes more complicated in the coming cycles. 
Moreover no big organization is usign npm as its package manager. vercel, dub used pnpm, calcom uses yarn as their package managers.

File: src/components/VideoPlayer2.tsx [hunk 1]
```diff
@@ -77,40 +77,40 @@ export const VideoPlayer: FunctionComponent<VideoPlayerProps> = ({
         const newIndexDown =
           currentIndexDown !== 0 ? currentIndexDown - 1 : currentIndexDown;
         switch (event.code) {
-        case 'Period': // Increase playback speed
```
Comment (hkirat) [inline src/components/VideoPlayer2.tsx:80]: Is this just noise?
Or is there any real change here?

File: src/components/VideoPlayer.tsx [hunk 2]
```diff
@@ -22,23 +22,23 @@ export const VideoPlayer = ({
     }
     const handleKeyPress = (event: any) => {
       switch (event.code) {
-      case 'Space': // Space bar for play/pause
-        if (player.paused()) {
```
Comment (hkirat) [inline src/components/VideoPlayer.tsx:26]: Same question here

**Review by hkirat — CHANGES_REQUESTED** [review]
just two comments then good to go

File: src/components/VideoPlayer.tsx (hunk 2 above)
Comment (rohithreddy009) [reply src/components/VideoPlayer.tsx:26]: these are just noise, no real change done from my side

**Comment by rohithreddy009** [comment]
> just two comments then good to go

Good to go ser  
I tested changes locally.

File: src/components/VideoPlayer.tsx (hunk 2 above)
Comment (rohithreddy009) [reply src/components/VideoPlayer.tsx:26]: > Same question here

This change fixes the bug of videoplayer controls not working on full screens.

**Pull request closed without merging**